### PR TITLE
fix(core): fix stuttering when change doc title

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#6269** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — APPROVED** [review]
[fthis](https://app.graphite.dev/settings/meme-library?org=toeverything)

File: packages/common/infra/src/page/record-list.ts [hunk 1]
```diff
@@ -31,7 +30,12 @@ export class PageRecordList {
       return () => {
         dispose();
       };
-    }),
+    }).pipe(
+      distinctUntilChanged((p, c) => isEqual(p, c)),
+      map(ids =>
+        ids.map(id => new PageRecord(id, this.workspace, this.localState))
```
Comment (pengx17) [inline packages/common/infra/src/page/record-list.ts:36]: Any idea on why recreating `PageRecord` is resource-intensive?

Comment (EYHN) [reply packages/common/infra/src/page/record-list.ts:36]: Because our pagelist is too long, there are 500+

**Pull request merged**